source/geh_calculated_measurements/src/geh_calculated_measurements/net_consumption_group_6/domain/cenc.py

```diff
@@ -23,11 +27,187 @@ def calculate_cenc(
     orchestration_instance_id: str,
     execution_start_datetime: datetime,
 ) -> Cenc:
-    """Return a data frame with schema `cenc_schema`."""
-    # TODO JVM: Replace this dummy code
-    spark = initialize_spark()
-    # TODO JVM: Hardcoded data to match the first scenario test
-    data = [("00000000-0000-0000-0000-000000000001", "150000001500170200", Decimal("1000.000"), 2025, 1)]
-    df = spark.createDataFrame(data, schema=Cenc.schema)
-
-    return Cenc(df)
+    """Calculate net energy consumption (CENC) for metering points.
+
+    Returns a DataFrame with schema `cenc_schema`.
+    """
+    # Constants for estimated consumption for move_in cases
+    ESTIMATED_CONSUMPTION_MOVE_IN = 1800
+    ESTIMATED_CONSUMPTION_MOVE_IN_WITH_HEATING = 5600
+
+    # Filter and join the data
+    filtered_time_series = _filter_relevant_time_series_points(time_series_points)
+    parent_child_joined = _join_parent_child_with_consumption(
+        child_metering_points, consumption_metering_point_periods, execution_start_datetime
+    )
+
+    # Add timestamp from settlement month
+    parent_child_joined = _add_timestamp_from_settlement_month(parent_child_joined, execution_start_datetime, time_zone)
+
+    # Extract net consumption points
+    net_consumption_points = _get_net_consumption_metering_points(parent_child_joined)
```

Comment:
"Points" can be both metering and time series.
```suggestion
    net_consumption_metering_points = _get_net_consumption_metering_points(parent_child_joined)
```